client/gutenberg/editor/utils.js

```diff
@@ -3,29 +3,37 @@
  * External dependencies
  */
 import apiFetch from '@wordpress/api-fetch';
+import wpcomProxyRequest from 'wpcom-proxy-request';
 import { noop } from 'lodash';
 
-/**
- * Internal dependencies
- */
-import debugFactory from 'debug';
-
-const debug = debugFactory( 'calypso:gutenberg' );
-
 export const overrideAPIPaths = siteSlug => {
-	// TODO: no API support for now. We'll also need to handle authorization here.
-	apiFetch.use( () => noop );
-
 	const rootURL = 'https://public-api.wordpress.com/';
 	apiFetch.use( apiFetch.createRootURLMiddleware( rootURL ) );
 
-	// rewrite default API paths to match WP.com equivalents
-	// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
 	apiFetch.use( ( options, next ) => {
-		const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
-
-		debug( 'sending API request to: ', wpcomPath );
-
-		return next( { ...options, path: wpcomPath } );
+		if ( options.path.indexOf( 'wp/v2/sites/' ) >= 0 ) {
+			// rewrite default API paths to match WP.com equivalents
+			// Example: /wp/v2/posts -> /wp/v2/sites/{siteSlug}/posts
+			const wpcomPath = `/wp/v2/sites/${ siteSlug }/` + options.path.replace( '/wp/v2/', '' );
+			return next( { ...options, path: wpcomPath } );
+		} else if ( options.path.indexOf( '/me' ) >= 0 ) {
+			// rewrite the namespace of /me endpoints
+			// example: /wp/v2/me/settings -> /rest/v1.1/me/settings
+			return new Promise( ( resolve, reject ) => {
+				wpcomProxyRequest(
```

Comment:
Aside, would be nice if `wpcom-proxy-request` was able to return a promise :slightly_frowning_face:

Reply:
🤷‍♂️